### PR TITLE
HTTP abstraction

### PR DESCRIPTION
**Pull request by twz123**
Concentrate all HTTP related code into NakadiHttpClient.

Add some early, simple adoption of HTTP response routing à la Riptide.

### CODE REVIEW

**Comment by fatroom** [comment]
👍

**Comment by ehrlichandreas** [comment]
👍

File: src/main/java/org/zalando/rxnakadi/http/AhcResponseDispatch.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+            final Integer statusCode = response.getStatusCode();
+            for (final Route<Response, ? super Integer, R> route : routes) {
+                if (route.selector().test(statusCode)) {
+                    return route.apply(response);
+                }
+            }
+
+            return Single.error(new UnsupportedOperationException(
+                        String.format("Unsupported %s %s: %s", "status code", statusCode, responseString(response))));
```
Comment (cberg-zalando) [inline src/main/java/org/zalando/rxnakadi/http/AhcResponseDispatch.java]: Any reason, why you not directly integrate `status code` into the format string?

**Comment by cberg-zalando** [comment]
Some more documentation for interfaces and classes would definitely help. :smile:

**Pull request merged**